src/gql/GQLWrapper.tsx

```diff
@@ -96,6 +96,12 @@ const cache = new InMemoryCache({
     User: {
       keyFields: ["userId"],
     },
+    UserSettings: {
+      keyFields: false,
+      merge(existing, incoming, { mergeObjects }) {
+        return mergeObjects(existing, incoming);
+      },
+    },
```

Comment:
What necessitated this? If it's a testing/storybook specific issue it would probably make sense to have a separate mock cache

Reply:
It was in response to [this](https://www.apollographql.com/docs/react/errors/#%7B%22version%22%3A%223.8.4%22%2C%22message%22%3A12%2C%22args%22%3A%5B%22spruceConfig%22%2C%22Query%22%2C%22either%20ensure%20all%20objects%20of%20type%20SpruceConfig%20have%20an%20ID%20or%20a%20custom%20merge%20function%2C%20or%20%22%2C%22Query.spruceConfig%22%2C%22%7B%5Cn%20%20%5C%22__typename%5C%22%3A%20%5C%22SpruceConfig%5C%22%2C%5Cn%20%20%5C%22banner%5C%22%3A%20%5C%22%5C%22%2C%5Cn%20%20%5C%22bannerTheme%5C%22%3A%20%5C%22warning%5C%22%2C%5Cn%20%20%5C%22containerPools%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22pools%5C%22%3A%20%5B%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20%5C%22distro%5C%22%3A%20%5C%22localhost%5C%22%2C%5Cn%20%20%20%20%20%20%20%20%5C%22id%5C%22%3A%20%5C%22test-pool%5C%22%2C%5Cn%20%20%20%20%20%20%20%20%5C%22maxContainers%5C%22%3A%205%2C%5Cn%20%20%20%20%20%20%20%20%5C%22port%5C%22%3A%201234%5Cn%20%20%20%20%20%20%7D%5Cn%20%20%20%20%5D%5Cn%20%20%7D%2C%5Cn%20%20%5C%22jira%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22__typename%5C%22%3A%20%5C%22JiraConfig%5C%22%2C%5Cn%20%20%20%20%5C%22email%5C%22%3A%20%5C%22test%40example.com%5C%22%2C%5Cn%20%20%20%20%5C%22host%5C%22%3A%20%5C%22jira.mongodb.org%5C%22%5Cn%20%20%7D%2C%5Cn%20%20%5C%22keys%5C%22%3A%20%5B%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22location%5C%22%3A%20%5C%22%2Fpath%2Fto%2Fkey%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22name%5C%22%3A%20%5C%22fake_key%5C%22%5Cn%20%20%20%20%7D%5Cn%20%20%5D%2C%5Cn%20%20%5C%22providers%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22__typename%5C%22%3A%20%5C%22CloudProviderConfig%5C%22%2C%5Cn%20%20%20%20%5C%22aws%5C%22%3A%20%7B%5Cn%20%20%20%20%20%20%5C%22__typename%5C%22%3A%20%5C%22AWSConfig%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22maxVolumeSizePerUser%5C%22%3A%201500%2C%5Cn%20%20%20%20%20%20%5C%22pod%5C%22%3A%20null%5Cn%20%20%20%20%7D%5Cn%20%20%7D%2C%5Cn%20%20%5C%22slack%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22name%5C%22%3A%20%5C%22everygreen_slack%5C%22%5Cn%20%20%7D%2C%5Cn%20%20%5C%22spawnHost%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22__typename%5C%22%3A%20%5C%22SpawnHostConfig%5C%22%2C%5Cn%20%20%20%20%5C%22spawnHostsPerUser%5C%22%3A%206%2C%5Cn%20%20%20%20%5C%22unexpirableHostsPerUser%5C%22%3A%202%2C%5Cn%20%20%20%20%5C%22unexpirableVolumesPerUser%5C%22%3A%201%5Cn%20%20%7D%2C%5Cn%20%20%5C%22ui%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22__typename%5C%22%3A%20%5C%22UIConfig%5C%22%2C%5Cn%20%20%20%20%5C%22defaultProject%5C%22%3A%20%5C%22evergreen%5C%22%5Cn%20%20%7D%5Cn%7D%22%2C%22%7B%5Cn%20%20%5C%22banner%5C%22%3A%20%5C%22%5C%22%2C%5Cn%20%20%5C%22bannerTheme%5C%22%3A%20%5C%22warning%5C%22%2C%5Cn%20%20%5C%22containerPools%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22pools%5C%22%3A%20%5B%5Cn%20%20%20%20%20%20%7B%5Cn%20%20%20%20%20%20%20%20%5C%22distro%5C%22%3A%20%5C%22localhost%5C%22%2C%5Cn%20%20%20%20%20%20%20%20%5C%22id%5C%22%3A%20%5C%22test-pool%5C%22%2C%5Cn%20%20%20%20%20%20%20%20%5C%22maxContainers%5C%22%3A%205%2C%5Cn%20%20%20%20%20%20%20%20%5C%22port%5C%22%3A%201234%5Cn%20%20%20%20%20%20%7D%5Cn%20%20%20%20%5D%5Cn%20%20%7D%2C%5Cn%20%20%5C%22jira%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22email%5C%22%3A%20%5C%22test%40example.com%5C%22%2C%5Cn%20%20%20%20%5C%22host%5C%22%3A%20%5C%22jira.mongodb.org%5C%22%5Cn%20%20%7D%2C%5Cn%20%20%5C%22keys%5C%22%3A%20%5B%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22location%5C%22%3A%20%5C%22%2Fpath%2Fto%2Fkey%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22name%5C%22%3A%20%5C%22fake_key%5C%22%5Cn%20%20%20%20%7D%5Cn%20%20%5D%2C%5Cn%20%20%5C%22providers%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22aws%5C%22%3A%20%7B%5Cn%20%20%20%20%20%20%5C%22maxVolumeSizePerUser%5C%22%3A%201500%2C%5Cn%20%20%20%20%20%20%5C%22pod%5C%22%3A%20null%5Cn%20%20%20%20%7D%5Cn%20%20%7D%2C%5Cn%20%20%5C%22slack%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22name%5C%22%3A%20%5C%22everygreen_slack%5C%22%5Cn%20%20%7D%2C%5Cn%20%20%5C%22spawnHost%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22spawnHostsPerUser%5C%22%3A%206%2C%5Cn%20%20%20%20%5C%22unexpirableHostsPerUser%5C%22%3A%202%2C%5Cn%20%20%20%20%5C%22unexpirableVolumesPerUser%5C%22%3A%201%5Cn%20%20%7D%2C%5Cn%20%20%5C%22ui%5C%22%3A%20%7B%5Cn%20%20%20%20%5C%22defaultProject%5C%22%3A%20%5C%22evergreen%5C%22%5Cn%20%20%7D%5Cn%7D%22%5D%7D) warning which seemed valid. Since SpruceConfig does not have an id field it will not be properly cached and resolved

Reply:
Right now we only fetch SpruceConfig in one query but if we were to break it up to do [this](https://jira.mongodb.org/browse/DEVPROD-2984) Apollo would invalidate the old query everytime

Reply:
Hm that link doesn't seem to flag an error in `UserSettings` does it? How exactly does this solve the `SpruceConfig` error?

Reply:
🤦 wow what a complete fail on my part, I somehow misconstrued the two. I can remove this from the pr but I think it is still a valid change.

Reply:
Hahah, I actually think assigning another field like `slackMemberId` as the keyField may be a better fix here since that is a unique value. Or feel to omit and ticket!

Reply:
`slackMemberId` wouldn't really work for this. It can be nullable and isn't something that we would intentionally request every time we want to request `spruceConfig` like we would for something like a task or version id. (We could but it doesn't seem like it would make sense). I'll omit it from this PR and ticket it.